### PR TITLE
Opti log2

### PR DESCRIPTION
**Pull request by makcandrov**

### CODE REVIEW

**Review by Rubilmax — APPROVED** [review]
CI is failing for unknown permission issues
Tests are passing, and the gas diff is available in the CI output

File: src/math/Math.sol [hunk 1]
```diff
@@ -33,38 +33,36 @@ library Math {
...
-            mstore(m, 0x0001020903110a19042112290b311a3905412245134d2a550c5d32651b6d3a75)
-            mstore(add(m, 0x20), 0x06264262237d468514804e8d2b95569d0d495ea533a966b11c886eb93bc176c9)
-            mstore(add(m, 0x40), 0x071727374353637324837e9b47af86c7155181ad4fd18ed32c9096db57d59ee3)
-            mstore(add(m, 0x60), 0x0e2e4a6a5f92a6be3498aae067ddb2eb1d5989b56fd7baf33ca0c2ee77e5caf7)
-            mstore(add(m, 0x80), 0xff0810182028303840444c545c646c7425617c847f8c949c48a4a8b087b8c0c8)
-            mstore(add(m, 0xa0), 0x16365272829aaec650acd0d28fdad4e22d6991bd97dfdcea58b4d6f29fede4f6)
-            mstore(add(m, 0xc0), 0xfe0f1f2f3f4b5b6b607b8b93a3a7b7bf357199c5abcfd9e168bcdee9b3f1ecf5)
-            mstore(add(m, 0xe0), 0xfd1e3e5a7a8aa2b670c4ced8bbe8f0f4fc3d79a1c3cde7effb78cce6facbf9f8)
-            mstore(0x40, add(m, 0x100)) // Update the free memory pointer
+            z := shl(7, gt(x, 0xffffffffffffffffffffffffffffffff))
```
Comment (MerlinEgalite) [inline src/math/Math.sol:40]:
```suggestion
            let z := shl(7, gt(x, 0xffffffffffffffffffffffffffffffff))
```

**Review by MerlinEgalite — APPROVED** [review]

**Review by QGarchery — COMMENTED** [review]
Great work ! I think we should remove the de Bruijn library too, and add some comments, so I opened #95. I would rather that the PR I just opened points to this one but I don't know how to do it or if it is possible: I don't have the rights on the public forked repo here https://github.com/meppent/morpho-utils 

See also #94 if you want (not important)

**Comment by MathisGD** [comment]
Closing, in favor of #95

**Pull request closed without merging**